lib/dockerregistry/inventory.go

```diff
@@ -1952,6 +1952,38 @@ func MKPopulateRequestsForPromotionEdges(
 	}
 }
 
+// CheckPromotion is where PreChecks to be run are defined.
+func (sc *SyncContext) CheckPromotion(
+	edges map[PromotionEdge]interface{},
+) error {
+	checksToRun := []PreCheck{}
+	return sc.RunChecks(edges, checksToRun)
+}
+
+// RunChecks runs defined PreChecks in order to check the promotion.
+func (sc *SyncContext) RunChecks(
+	edges map[PromotionEdge]interface{},
+	preChecks []PreCheck,
+) error {
+
+	var errors []error
+	for _, preCheck := range preChecks {
+		err := preCheck.Run(edges)
+		if err != nil {
+			errors = append(errors, err)
+		}
+	}
+
+	if errors != nil {
+		for _, err := range errors {
```

Comment:
You can just log the errors as you append them in the for-loop above this. Then you only need 1 for-loop instead of 2.